workers/github_worker_base.rb

```diff
@@ -58,11 +58,17 @@ def thread_id
     # This is a separate method, so it's lazy loaded
     # since it will be run on the "main" thread if it were
     # part of the #initialize method
-    def git_repo
-      @git_repo ||= GitRepo.new(
-        git_config: project.repo_config,
-        provider_credential: provider_credential
-      )
+    def git_repo_service
```

Comment:
Gonna need some help to refactor this class, PRs into my branch are more than welcome 😄

Reply:
@taquitos